server/api.js

```diff
@@ -51,7 +47,30 @@ const auth = async (req, res, next) => {
 		}
 
 		req.user = user;
-		next();
+
+		if (user.role === "admin") {
+			return next();
+		}
+
+		if (method === "POST" && path === "/tender") {
+			if (user.role === "buyer") {
+				return next();
+			}
+			return res.status(403).json({ code: "FORBIDDEN" });
+		}
+
+		if (method === "GET") {
+			const publicPaths = [
+				"/skills",
+				"/buyer-tender",
+				"/bidder-bid",
+				"/tenders",
+			];
+			if (publicPaths.includes(path)) {
+				return next();
+			}
+		}
+		return res.status(403).json({ code: "FORBIDDEN" });
```

Comment:
This is basically what we want to do, but having a map that allows us to add routes without having to add more code is beneficial for the future.

Do something similar to what I suggested here for the map: https://github.com/LauraSantiag0/Love-Me-Tender/issues/77

And combine it with the logic you have here

Reply:
working on it